tests/integration/test_expose_external.py

```diff
@@ -0,0 +1,221 @@
+else:
+    TLS_APP_NAME = "tls-certificates-operator"
+    TLS_CHANNEL = "legacy/edge" if architecture == "arm64" else "legacy/stable"
+    TLS_CONFIG = {"generate-self-signed-certificates": "true", "ca-common-name": "Test CA"}
+
+
+async def confirm_cluster_ip_endpoints(ops_test: OpsTest) -> None:
+    """Helper function to test the cluster ip endpoints."""
+    for attempt in tenacity.Retrying(
+        reraise=True,
+        stop=tenacity.stop_after_delay(SLOW_TIMEOUT),
+        wait=tenacity.wait_fixed(10),
+    ):
+        with attempt:
+            data_integrator_unit = ops_test.model.applications[DATA_INTEGRATOR].units[0]
+            credentials = await get_data_integrator_credentials(data_integrator_unit)
+
+    assert credentials["postgresql"]["database"] == TEST_DATABASE_NAME, "Database is empty"
+    username = credentials["postgresql"]["username"]
+    assert username is not None, "Username is empty"
+    password = credentials["postgresql"]["password"]
+    assert password is not None, "Password is empty"
+
+    endpoint_name = f"pgbouncer-k8s-service.{ops_test.model.name}.svc.cluster.local"
+    assert credentials["postgresql"]["endpoints"] == f"{endpoint_name}:6432", (
+        "Endpoint is unexpected"
+    )
+
+    assert (
+        credentials["postgresql"]["uris"]
+        == f"postgresql://{username}:{password}@{endpoint_name}:6432/{TEST_DATABASE_NAME}"
+    ), "URIs is unexpected"
+
+
+async def confirm_endpoint_connectivity(ops_test: OpsTest) -> None:
+    """Helper to confirm endpoint connectivity."""
+    for attempt in tenacity.Retrying(
+        reraise=True,
+        stop=tenacity.stop_after_delay(SLOW_TIMEOUT),
+        wait=tenacity.wait_fixed(10),
+    ):
+        with attempt:
+            data_integrator_unit = ops_test.model.applications[DATA_INTEGRATOR].units[0]
+            credentials = await get_data_integrator_credentials(data_integrator_unit)
+            assert credentials["postgresql"]["uris"] is not None, "URIs is missing"
+
+            database = credentials["postgresql"]["database"]
+            user = credentials["postgresql"]["username"]
+            password = credentials["postgresql"]["password"]
+            endpoints = credentials["postgresql"]["endpoints"]
+            host, port = endpoints.split(",")[0].split(":")
+
+            with psycopg2.connect(
+                f"dbname='{database}' user='{user}' password='{password}' host='{host}' port='{port}' connect_timeout=10"
+            ) as connection, connection.cursor() as cursor:
+                cursor.execute("SELECT 1;")
+                assert cursor.fetchone()[0] == 1, "Unable to execute query"
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_expose_external(ops_test) -> None:
+    """Test the expose-external config option."""
+    logger.info("Building pgbouncer-k8s charm")
+    pgbouncer_charm = await ops_test.build_charm(".")
+    # await ops_test.model.set_config(MODEL_CONFIG)
+
```

Comment:
Is this still relevant here?

Reply:
yes, still relevant. uncommented in [8b61721](https://github.com/canonical/pgbouncer-k8s-operator/pull/495/commits/8b617217a7b111fb873a685f5bdd3850f499078d)